sklearn/tests/test_impute.py

```diff
@@ -309,3 +313,215 @@ def test_imputation_copy():
 
     # Note: If X is sparse and if missing_values=0, then a (dense) copy of X is
     # made, even if copy=False.
+
+
+def test_mice_rank_one():
+    rng = np.random.RandomState(0)
+    d = 100
+    A = rng.rand(d, 1)
+    B = rng.rand(1, d)
+    X = np.dot(A, B)
+    nan_mask = rng.rand(d, d) < 0.5
+    X_missing = X.copy()
+    X_missing[nan_mask] = np.nan
+
+    imputer = MICEImputer(n_imputations=5,
+                          n_burn_in=5,
+                          verbose=True)
+    X_filled = imputer.fit_transform(X_missing)
+    assert_array_almost_equal(X_filled, X, decimal=2)
+
+
+def test_mice_imputation_order():
+    rng = np.random.RandomState(0)
+    n = 100
+    d = 10
+    X = sparse_random_matrix(n, d, density=0.10,
+                             random_state=rng).toarray()
+    X[:, 0] = 1  # this column shouldn't be ever used
+
+    for imputation_order in ['random', 'roman', 'monotone',
```

Comment:
We can use `pytest.mark.parametrize` to have a cleaner test. Pass the order as an argument of the function.
Depending of the type you might want to pass a second argument to make a single assert if this is possible.

Reply:
I don't think I can make a single assert here. The logic is pretty convoluted for each type of `imputation_order`